lib/src/fsmonitor.rs

```diff
@@ -44,20 +48,27 @@ pub enum FsmonitorKind {
     None,
 }
 
-impl FromStr for FsmonitorKind {
-    type Err = config::ConfigError;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "watchman" => Ok(Self::Watchman),
-            "test" => Err(config::ConfigError::Message(
-                "cannot use test fsmonitor in real repository".to_string(),
-            )),
-            "none" => Ok(Self::None),
-            other => Err(config::ConfigError::Message(format!(
-                "unknown fsmonitor kind: {}",
-                other
-            ))),
+impl FsmonitorKind {
+    /// Creates an `FsmonitorKind` from a `config`.
+    pub fn from_config(config: &Config) -> Result<FsmonitorKind, ConfigError> {
+        match config.get_string("core.fsmonitor") {
+            Ok(s) => match s.as_str() {
+                "watchman" => Ok(Self::Watchman {
+                    set_up_trigger: config
+                        .get_bool("core.watchman.set_up_trigger")
+                        .unwrap_or_default(),
+                }),
+                "test" => Err(ConfigError::Message(
+                    "cannot use test fsmonitor in real repository".to_string(),
+                )),
+                "none" => Ok(Self::None),
+                other => Err(ConfigError::Message(format!(
+                    "unknown fsmonitor kind: {}",
+                    other
+                ))),
```

Comment:
```suggestion
                other => Err(ConfigError::Message(format!(
                    "unknown fsmonitor kind: {other}",
                ))),
```

suggestion: I'm pretty sure our minimum-supported Rust version supports interpolated format strings, so you can use them here.